[chrome.angle] Force Pure RISCV64 Support

McKnight22 opened the pull request.

McKnight22 committed:
[Android] Force Pure RISCV64 Support

Signed-off-by: McKnight22 <tao.wang.2261@gmail.com>

unicornx commented:
I changed the tile of this PR to keep align with other PRs for chrome porting work, please check it out.

unicornx commented on android/angle_apk.gni:
```diff
@@ -46,14 +46,14 @@ if (enable_java_templates) {
       if (symbol_level != 0) {
         deps += [ ":compressed_symbols" ]
-        if (android_64bit_target_cpu) {
+        if (android_64bit_target_cpu && !invoker.skip_secondary_abi) {
```
Please clarify why you find out these two changes.

McKnight22 replied:
> Please clarify why you find out these two changes.

For these 2 changes, it's due to android_secondary_abi_toolchain introduced here and android_secondary_abi_toolchain should be skipped for Android RISCV64 target to enable pure RISCV64 support.

McKnight22 committed:
[chrome.angle] Force Pure RISCV64 Support - PatchSet.1

utilize the definition of skip_secondary_abi_for_cq and android_secondary_abi_cpu directly for Android RISCV64

Signed-off-by: McKnight22 <tao.wang.2261@gmail.com>

unicornx commented on android/angle_apk.gni:
```diff
@@ -46,14 +46,14 @@ if (enable_java_templates) {
       if (symbol_level != 0) {
         deps += [ ":compressed_symbols" ]
-        if (android_64bit_target_cpu) {
+        if (android_64bit_target_cpu && defined(android_secondary_abi_cpu)) {
```
2nd review:
建议用 `android_secondary_abi_toolchain`, 和下面代码结合比较紧密，好理解。

unicornx commented on android/angle_apk.gni:
```diff
@@ -46,14 +46,14 @@ if (enable_java_templates) {
       uncompress_shared_libraries = true
 
-      if (android_64bit_target_cpu) {
+      if (android_64bit_target_cpu && defined(android_secondary_abi_cpu)) {
```
2nd review:
建议用 `android_secondary_abi_toolchain`, 和下面代码结合比较紧密，好理解。

unicornx commented:
先关闭这个 pr，新建 https://github.com/aosp-riscv/angle/pull/5 作为总结性提交。
